### PR TITLE
fix: Credential syncing - ensure response data is always returned.

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

Currently, our apps handle the response of their refresh token requests in different ways, either with the response directly or the response body. This PR aims to have the token refresh requests to always return the response body.

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is a UI/UX design document, please, share it here.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- For each app changed in this PR, ensure that the tokens are being refreshed properly

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/app-store/_utils/oauth/refreshOAuthTokens.ts [hunk 1]
```diff
@@ -25,7 +25,15 @@ const refreshOAuthTokens = async (refreshFunction: () => any, appSlug: string, u
...
-    return response;
+
+    if (!response.ok) {
+      throw new Error(
+        `Failed to refresh OAuth tokens for ${appSlug} userId ${userId}: ${response.statusText}`
+      );
+    }
+
+    const responseBody = await response.json();
+    return responseBody;
```
Comment (joeauyeung) [inline packages/app-store/_utils/oauth/refreshOAuthTokens.ts:36]: When credential syncing, return the body of the payload from the credential sync endpoint

File: packages/app-store/zoho-bigin/lib/CalendarService.ts [hunk 2]
```diff
@@ -83,31 +83,35 @@ export default class BiginCalendarService implements Calendar {
     };
 
     const tokenInfo = await refreshOAuthTokens(
-      async () =>
-        await axios.post(accountsUrl, qs.stringify(formData), {
+      async () => {
+        const response = await fetch(accountsUrl, {
```
Comment (joeauyeung) [inline packages/app-store/zoho-bigin/lib/CalendarService.ts:87]: Axios has the `data` property but not the `body`. To ensure compatibility with credential syncing, let's use the native fetch method.

File: packages/app-store/zoho-bigin/lib/CalendarService.ts [hunk 3]
```diff
@@ -83,31 +83,35 @@ export default class BiginCalendarService implements Calendar {
...
         where: {
           id: credentialId,
         },
         data: {
-          key: tokenInfo.data as BiginToken,
+          key: tokenInfo.body as BiginToken,
         },
       });
-      this.log.debug("Fetched token", tokenInfo.data.access_token);
+      this.log.debug("Fetched token", tokenInfo.body.access_token);
```
Comment (Udit-takkar) [inline packages/app-store/zoho-bigin/lib/CalendarService.ts:114]: L119 in this file returns tokenInfo.data instead of tokenInfo.body

**Comment by keithwillcode** [comment]
@joeauyeung is this PR still valid now that #14588 went in?

**Comment by joeauyeung** [comment]
Close in favour of https://github.com/calcom/cal.com/pull/14588

**Pull request closed without merging**